From f13080d8abfd16b2cc44c3d83e17456872d7a169 Mon Sep 17 00:00:00 2001
From: Richard Su <rwsu@redhat.com>
Date: Tue, 25 Mar 2014 12:03:34 -0700
Subject: [PATCH 5/5] Update nova's selinux policies

Nova needs permissions to /mnt/state/var/lib/nova and
/mnt/state/var/log/nova.

Change-Id: Ie1f23d5d11b725c77648b15ad0738f30ada36a83
---
 elements/nova/os-refresh-config/configure.d/10-nova-state   |  2 ++
 elements/nova/os-refresh-config/configure.d/20-nova-selinux | 10 ++++++++++
 2 files changed, 12 insertions(+)
 create mode 100755 elements/nova/os-refresh-config/configure.d/20-nova-selinux

diff --git a/elements/nova/os-refresh-config/configure.d/10-nova-state b/elements/nova/os-refresh-config/configure.d/10-nova-state
index cd738b1..d304646 100755
--- a/elements/nova/os-refresh-config/configure.d/10-nova-state
+++ b/elements/nova/os-refresh-config/configure.d/10-nova-state
@@ -1,3 +1,5 @@
 #!/bin/bash
 [ -d /mnt/state/var/lib/nova ] && exit 0
 install -D -d -o nova -g nova -m 0775 /mnt/state/var/lib/nova
+[ -d /mnt/state/var/log/nova ] && exit 0
+install -D -d -o nova -g nova -m 0775 /mnt/state/var/log/nova
diff --git a/elements/nova/os-refresh-config/configure.d/20-nova-selinux b/elements/nova/os-refresh-config/configure.d/20-nova-selinux
new file mode 100755
index 0000000..5eb322b
--- /dev/null
+++ b/elements/nova/os-refresh-config/configure.d/20-nova-selinux
@@ -0,0 +1,10 @@
+#!/bin/bash
+set -eu
+
+[ -x /usr/sbin/semanage ] || exit 0
+
+semanage fcontext -a -t nova_var_lib_t "/mnt/state/var/lib/nova(/.*)?"
+restorecon -Rv /mnt/state/var/lib/nova
+
+semanage fcontext -a -t nova_log_t "/mnt/state/var/log/nova(/.*)?"
+restorecon -Rv /mnt/state/var/log/nova
-- 
1.9.0

